src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+
+  get height() {
+    return this.framebuffer.height * this.framebuffer.density;
+  }
+
+  rawTexture() {
+    return this.framebuffer[this.property];
+  }
+}
+
+p5.FramebufferTexture = FramebufferTexture;
+
+class Framebuffer {
+  /**
+   * @class p5.Framebuffer
+   * @constructor
+   * @param {p5.Graphics|p5} target A p5 global instance or p5.Graphics
+   * @param {Object} [settings] A settings object
+   */
+  constructor(target, settings = {}) {
+    this.target = target;
+    this.target._renderer.framebuffers.add(this);
+
+    this.format = settings.format || constants.UNSIGNED_BYTE;
+    this.channels = settings.channels || (
+      target._renderer._pInst._glAttributes.alpha
+        ? constants.RGBA
+        : constants.RGB
+    );
+    this.depthFormat = settings.depthFormat || (
+      this.format === constants.UNSIGNED_BYTE
+        ? constants.UNSIGNED_BYTE
+        : constants.FLOAT
+    );
+    this.antialias = (
+      settings.antialias === undefined
+        ? target._pInst._glAttributes.antialias
+        : settings.antialias
+    );
+    if (this.antialias && target.webglVersion !== constants.WEBGL2) {
+      console.warn('Antialiasing is unsupported in a WebGL 1 context');
+      this.antialias = false;
+    }
+    if (settings.width && settings.height) {
+      this.width = settings.width;
+      this.height = settings.height;
+      this.autoSized = false;
+    } else {
+      this.width = target.width;
+      this.height = target.height;
+      this.autoSized = true;
+    }
+    this.density = settings.density || target.pixelDensity();
+
+    const gl = target._renderer.GL;
+    this.gl = gl;
+
+    this._checkIfFormatsAvailable();
+
+    this.framebuffer = gl.createFramebuffer();
+    if (!this.framebuffer) {
+      throw new Error('Unable to create a framebuffer');
+    }
+    if (this.antialias) {
+      this.aaFramebuffer = gl.createFramebuffer();
+      if (!this.aaFramebuffer) {
+        throw new Error('Unable to create a framebuffer');
+      }
+    }
+
+    this._recreateTextures();
+
+    const prevCam = this.target._renderer._curCamera;
+    this.defaultCamera = this.createCamera();
+    this.target._renderer._curCamera = prevCam;
+  }
+
+  resizeCanvas(width, height) {
+    this.autoSized = false;
+    this.width = width;
+    this.height = height;
+    this._handleResize();
+  }
+
+  pixelDensity(density) {
+    if (density) {
+      this.autoSized = false;
+      this.density = density;
+      this._handleResize();
+    } else {
+      return this.density;
+    }
+  }
+
+  autoSized(autoSized) {
+    if (autoSized === undefined) {
+      return this.autoSized;
+    } else {
+      this.autoSized = autoSized;
+      this._handleResize();
+    }
+  }
+
+  _checkIfFormatsAvailable() {
+    const gl = this.gl;
+
+    if (
+      this.target.webglVersion === constants.WEBGL &&
+      !gl.getExtension('WEBGL_depth_texture')
+    ) {
+      throw new Error('Unable to create depth textures in this environment');
```

Comment:
Do you think it's worth exposing if depth texture capabilities are available to the end user? I could see it being helpful for someone who wants to gracefully fallback to something else if they're not supported.

Reply:
How do you feel about making this not throw an error, but just leaves `fbo.depth` undefined? So in that case one would make a framebuffer and then check if its depth property exists.

Reply:
sgtm, though I would leave the error in place as well!

Reply:
Sounds good, maybe just as a warning?